common/rpc/interceptor/telemetry.go

```diff
@@ -67,6 +68,22 @@ func NewTelemetryInterceptor(
 	}
 }
 
+
+// Use this method to override scope used for reporting a metric.
+// Ideally this method should never be used.
+func (ti *TelemetryInterceptor) overrideScope(scope int, methodName string, req interface{}) int {
+	// GetWorkflowExecutionHistory method handles both long poll and regular calls.
+	// Current plan is to eventually split GetWorkflowExecutionHistory into two APIs,
+	// remove this if case when that is done.
+	if scope == metrics.FrontendGetWorkflowExecutionHistoryScope && methodName == "GetWorkflowExecutionHistory" {
```

Comment:
this check `scope == metrics.FrontendGetWorkflowExecutionHistoryScope` is unnecessary

Reply:
It is not explicitly required, but I prefer to add it for safety in case of surrounding logic change.

Reply:
then remove `methodName == "GetWorkflowExecutionHistory"`?

Reply:
the mapping is already been unit tested
https://github.com/temporalio/temporal/blob/master/common/metrics/api_metrics_test.go

Reply:
done.